### PR TITLE
feat(sfs): add a new datasource to get the list of permission rules

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:

Add a new datasource to get the list of permission rules.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1.support a new datasource to get the list of the permission rules.
2.support related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.

```
$ make testacc TEST="./huaweicloud/services/acceptance/sfs" TESTARGS="-run TestAccDataSourcePermRules_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/sfs -v -run TestAccDataSourcePermRules_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourcePermRules_basic
=== PAUSE TestAccDataSourcePermRules_basic
=== CONT  TestAccDataSourcePermRules_basic
--- PASS: TestAccDataSourcePermRules_basic (391.16s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/sfs       391.239s
```

* [x] Documentation updated.
* [x] Schema updated.
* [ ] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

File: docs/data-sources/sfs_turbo_perm_rules.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+---
+subcategory: "Scalable File Service (SFS)"
+layout: "huaweicloud"
+page_title: "HuaweiCloud: huaweicloud_sfs_turbo_perm_rules"
+description: |-
+  Use this datasource to get list of the permission rules.
```
Comment (deer-hang) [inline docs/data-sources/sfs_turbo_perm_rules.md]:
```suggestion
  Use this datasource to get a list of permission rules.
```

File: docs/data-sources/sfs_turbo_perm_rules.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+---
+
+# huaweicloud_sfs_turbo_perm_rules
+
+Use this datasource to get list of the permission rules.
```
Comment (deer-hang) [inline docs/data-sources/sfs_turbo_perm_rules.md]:
```suggestion
Use this datasource to get a list of permission rules.
```

File: docs/data-sources/sfs_turbo_perm_rules.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+* `rw_type` - The read and write permission of the authorized object.
+  The value can be **rw** (read and write), **ro** (read only) or **none** (no permission).
+  The default value is **rw**.
+
+* `user_type` - The file system access permission granted to the user of the authorized object.
+  The valid values are as follow:
+  + **no_root_squash**: Allow the root user on the client to access the file system as root.
+  + **root_squasg**: Allow the root user on the client to access the file system as anonymous (nfsnobody).
+  + **all_squash**: Allow any user on the client to access the file system as nfsnobody.
+  The default value is **all_squash**.
```
Comment (deer-hang) [inline docs/data-sources/sfs_turbo_perm_rules.md:59]:
```suggestion

  The default value is **all_squash**.
```

File: huaweicloud/services/acceptance/sfs/data_source_huaweicloud_sfs_turbo_perm_rules_test.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttrSet(dataSource, "rules.#"),
+					resource.TestCheckResourceAttrSet(dataSource, "rules.0.id"),
+					resource.TestCheckResourceAttrSet(dataSource, "rules.0.ip_cidr"),
+					resource.TestCheckResourceAttrSet(dataSource, "rules.0.rw_type"),
+					resource.TestCheckResourceAttrSet(dataSource, "rules.0.user_type"),
+
+					dc.CheckResourceExists(),
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/sfs/data_source_huaweicloud_sfs_turbo_perm_rules_test.go:36]: remove this line.

File: huaweicloud/services/acceptance/sfs/data_source_huaweicloud_sfs_turbo_perm_rules_test.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+					resource.TestCheckOutput("is_exist_rule", "true"),
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/sfs/data_source_huaweicloud_sfs_turbo_perm_rules_test.go]: It is unnecessary to repeatedly check whether the rule exists.
Remove the extra `dc.CheckResourceExists()`

**Comment by ruwenqiang123** [comment]
```
$ make testacc TEST="./huaweicloud/services/acceptance/sfs" TESTARGS="-run TestAccDataSourcePermRules_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/sfs -v -run TestAccDataSourcePermRules_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourcePermRules_basic
=== PAUSE TestAccDataSourcePermRules_basic
=== CONT  TestAccDataSourcePermRules_basic
--- PASS: TestAccDataSourcePermRules_basic (442.69s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/sfs       442.737s
```

**Pull request merged**